GenericTrigger.lua

```diff
@@ -349,10 +368,39 @@ function WeakAuras.ScanEvents(event, arg1, arg2, ...)
       local updateTriggerState = false;
       for triggernum, data in pairs(triggers) do
         if(data.triggerFunc) then
-          if(data.triggerFunc(event, arg1, arg2, ...)) then
-            WeakAuras.ActivateEvent(id, triggernum, data);
-            updateTriggerState = true;
+          local untriggerCheck = false;
```

Comment:
So this slightly extends how trigger functions work. Trigger functions can declare in their prototype that their first parameter should be the trigger states. Either "all", or the "one" if they only need one state. 

If they have a statesParameter "one", they can store arguments into the state by just adding a "store = true" to that argument, that is what the unit characteristics trigger does.

If they have a statesParamter "all", they also need to have a variable "cloneId", which is the state they want to modify. 

Some of the later patches in the series make use of those new modes too.